addons/point_of_sale/models/pos_session.py

```diff
@@ -29,8 +29,7 @@ def _confirm_orders(self):
                 if order.state not in ('paid'):
                     raise UserError(_("You cannot confirm all orders of this session, because they have not the 'paid' status"))
                 order.action_pos_order_done()
-            orders_to_reconcile = session.order_ids.filtered(lambda order: order.state in ['invoiced', 'done'] and order.partner_id)
```

Comment:
I would simply filter here:
``` python
orders_to_reconcile = session.order_ids._filtered_for_reconciliation()
```

No need to move the `sudo` or filter somewhere else.